sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -31,65 +31,98 @@
+            str(key) + '=' + str(value) for key, value in params))
+        func_lines = [func_definition]
+        for i, d in enumerate(data_args):
+            line = '  {} = sdc.hiframes.api.init_series(sdc.hiframes.pd_dataframe_ext.get_dataframe_data(df, {}))'
+            func_lines.append(line.format(d + '_S', i))
+            func_lines.append('  {} = {}.{}({})'.format(d + '_O', d + '_S', name, ", ".join(
+                str(key) for key, value in params)))
+        func_lines.append("  return sdc.hiframes.pd_dataframe_ext.init_dataframe({}, None, {})\n".format(
+            ", ".join(d + '_O._data' for d in data_args),
+            ", ".join("'" + c + "'" for c in saved_columns)))
+
+        loc_vars = {}
+        func_text = '\n'.join(func_lines)
+        exec(func_text, {'sdc': sdc, 'np': numpy}, loc_vars)
+        _reduce_impl = loc_vars['_reduce_impl']
+
+        return _reduce_impl
+
+    def check_type(name, df, axis=None, skipna=None, level=None, numeric_only=None, ddof=1, min_count=0):
+        ty_checker = TypeChecker('Method {}().'.format(name))
+        ty_checker.check(df, DataFrameType)
+
+        if not (isinstance(axis, types.Omitted) or axis is None):
+            ty_checker.raise_exc(axis, 'unsupported', 'axis')
+
+        if not (isinstance(skipna, (types.Omitted, types.NoneType, types.Boolean)) or skipna is None):
+            ty_checker.raise_exc(skipna, 'bool', 'skipna')
+
+        if not (isinstance(level, types.Omitted) or level is None):
+            ty_checker.raise_exc(level, 'unsupported', 'level')
+
+        if not (isinstance(numeric_only, types.Omitted) or numeric_only is None):
+            ty_checker.raise_exc(numeric_only, 'unsupported', 'numeric_only')
+
+        if not (isinstance(ddof, types.Omitted) or ddof == 1):
+            ty_checker.raise_exc(ddof, 'unsupported', 'ddof')
+
+        if not (isinstance(min_count, types.Omitted) or min_count == 0):
+            ty_checker.raise_exc(min_count, 'unsupported', 'min_count')
+
+    @overload_method(DataFrameType, 'head')
+    def head_overload(df, n=5):
+        """
+        Pandas DataFrame method :meth:`pandas.DataFrame.head` implementation.
+        .. only:: developer
+            Test: python -m sdc.runtests sdc.tests.test_dataframe.TestDataFrame.test_head1
+        Parameters
+        -----------
+        self: :class:`pandas.DataFrame`
+            input arg
+        n: :obj:`int`, default 5
+            input arg, default 5
+        Returns
+        -------
+        :obj:`pandas.Series`
+        returns: The first n rows of the caller object.
+        """
+
+        name = 'head'
+
+        check_type(name, df)
+
+        ty_checker = TypeChecker('Method {}().'.format(name))
+        ty_checker.check(df, DataFrameType)
+
+        if not (isinstance(n, (types.Omitted, types.Integer)) or n == 5):
+            ty_checker.raise_exc(n, 'int64', 'n')
+
+        params = [('n', 5)]
```

Comment:
You could pass `[('n', 5)]` to the `sdc_pandas_dataframe_reduce_columns_series` directly without creating variable `params`.